### PR TITLE
feat(node-server-sdk) initial generator output and npm publish

### PR DESCRIPTION
**Pull request by tomer-friedman**
Created library `node-server-sdk` with generator - I put it under `libs`, but we might want a different directory for public publishable packages

Created github workflow to publish all `affected` packages to NPM registry after merge/push to master (so long as tests pass)

Still to do: Semversioning

### CODE REVIEW

File: .github/actions/npm-install-with-cache/action.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          ${{ steps.npm-cache-dir.outputs.dir }}
+          node_modules
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Install (on cache miss)
+      if: steps.npm-cache.outputs.cache-hit != 'true'
+      shell: bash
+      run: npm ci
```
Comment (tomer-friedman) [inline .github/actions/npm-install-with-cache/action.yml:25]: we can re-use these github actions in different flows

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      affected: ${{ steps.set-affected.outputs.affected }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v3
```
Comment (tomer-friedman) [inline .github/workflows/npm-publish.yml:20]: fetch depth 0 + nrwl/nx-set-shas@v3 are required for using affected. FYI for the future

File: tsconfig.base.json [hunk 3]
```diff
@@ -15,9 +15,8 @@
     "skipDefaultLibCheck": true,
     "baseUrl": ".",
     "paths": {
-      "@enrolla/enrolla-ts": ["libs/enrolla-ts/src/index.ts"],
-      "@enrolla/prisma-models": ["libs/prisma-models/src/index.ts"],
-      "@enrolla/ts-sdk": ["libs/enrolla-ts-sdk/src/index.ts"]
```
Comment (tomer-friedman) [inline tsconfig.base.json:20]: I assumed this wasnt needed

Comment (nirga) [reply tsconfig.base.json:20]: yep 👍

**Review by nirga — APPROVED** [review]

**Pull request merged**